sql/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner;
+
+import io.crate.analyze.QueriedTable;
+import io.crate.expression.eval.EvaluatingNormalizer;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RowGranularity;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.apache.lucene.util.BytesRef;
+import org.junit.Before;
+import org.junit.Test;
+
+import static io.crate.testing.SymbolMatchers.isLiteral;
+import static io.crate.testing.TestingHelpers.isDocKey;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+
+public class WhereClauseOptimizerTest extends CrateDummyClusterServiceUnitTest{
+
+    private SQLExecutor e;
+
+    @Before
+    public void setUpExecutor() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("create table bystring (name string primary key, score double) " +
+                      "clustered by (name) ")
+            .addTable("create table clustered_by_only (x int) clustered by (x)")
+            .addTable("create table tclustered (x int) clustered by (x)")
+            .addPartitionedTable(
+                "create table parted (id int, date timestamp) " +
+                "partitioned by (date)"
+            )
+            .addPartitionedTable(
+                "create table parted_pk (id int primary key, date timestamp primary key) " +
+                "partitioned by (date)",
+                new PartitionName("parted_pk", singletonList(new BytesRef("1395874800000"))).asIndexName(),
+                new PartitionName("parted_pk", singletonList(new BytesRef("1395961200000"))).asIndexName(),
+                new PartitionName("parted_pk", singletonList(null)).asIndexName()
+            )
+            .build();
+    }
+
+    private WhereClauseOptimizer.DetailedQuery optimize(String statement) {
+        QueriedTable queriedTable = e.analyze(statement);
+        EvaluatingNormalizer normalizer = new EvaluatingNormalizer(
+            e.functions(),
+            RowGranularity.CLUSTER,
+            null,
+            queriedTable.tableRelation()
+        );
+        return WhereClauseOptimizer.optimize(
+            normalizer,
+            queriedTable.where().queryOrFallback(),
+            ((DocTableInfo) queriedTable.tableRelation().tableInfo()),
+            e.getPlannerContext(clusterService.state()).transactionContext()
+        );
+    }
+
+    @Test
+    public void testFilterOn_IdDoesNotProduceDocKeysWithOnlyClusteredBy() {
```

Comment:
shouldn't this test be named `...WithoutClusteredBy` ?

Reply:
Renamed to `testFilterOn_IdDoesNotProduceDocKeysIfTableHasOnlyAClusteredByDefinition` - I think that was the purpose of the original test, but I'm not sure.